core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt

```diff
@@ -415,8 +415,14 @@ class DiplomacyManager() {
     fun makePeace(){
         diplomaticStatus= DiplomaticStatus.Peace
 
-        getCommonKnownCivs().forEach {
-            it.addNotification("[${civInfo.civName}] and [${otherCiv().civName}] have signed the Peace Treaty!", null, Color.WHITE)
+        if (otherCiv().isAtWarWith(civInfo)) {
```

Comment:
Couldn't think of a cleaner way to do it

Reply:
What's the point in checking `otherCiv().isAtWarWith(civInfo)`? If you are in `makePeace` it means a trade with a peace treaty has been signed, so the two civs must be at war.

Reply:
When AI makes peace with AI, there's this piece of code
```
        transferTrade(ourCivilization, otherCivilization, currentTrade)
        transferTrade(otherCivilization, ourCivilization, currentTrade.reverse())
```
which triggers the `makePeace()` twice.
I check the `otherCiv`s diplomacy manager to see if their side of peace deal is already done, and if it is, I don't push notification.
This avoids double-notification like "Russian made peace with America" and "America made peace with Russia"